react/src/pages/participants/ParticipantInfoDialog.tsx

```diff
@@ -49,38 +49,44 @@ const getParticipantValues = (participant: Participant) => {
         participant.updated_by,
     ];
 };
-function getAnalysisInfoList(analyses: Analysis[]): Info[] {
-    return analyses.map(analysis => {
-        return {
-            primaryListTitle: `Analysis ID ${analysis.analysis_id}`,
-            secondaryListTitle: `Current State: ${analysis.analysis_state} - Click for more details`,
-            titles: getAnalysisTitles(),
-            values: getAnalysisValues(analysis),
-        };
-    });
-}
 
 interface DialogProp {
     open: boolean;
     participant: Participant;
     onClose: () => void;
 }
 
+/**
+ * Given a participant, return a promise resolving to a list of all analyses
+ * of datasets tied to this participant.
+ */
+async function fetchAnalyses(participant: Participant) {
+    const datasets = participant.tissue_samples.flatMap(sample => sample.datasets);
+    let analysisList: Analysis[] = [];
+    for (const dataset of datasets) {
+        const response = await fetch("/api/datasets/" + dataset.dataset_id);
+        if (response.ok) {
+            const data = await response.json();
+            console.log(data);
+            analysisList = analysisList.concat(data.analyses as Analysis[]);
+        } else {
+            console.error(response);
+            throw new Error(`GET /api/datasets/${dataset.dataset_id} failed. Reason: ${response.status} - ${response.statusText}`);
```

Comment:
To be consistent with the rest, this should also be in the useEffect and enqueue a snackbar message if something goes wrong with the request.